pkg/scheduler/objects/queue.go

```diff
@@ -1050,14 +1069,22 @@ func (sq *Queue) sortApplications(filterApps bool) []*Application {
 	return sortApplications(sq.GetCopyOfApps(), queueSortType, sq.IsPrioritySortEnabled(), sq.GetGuaranteedResource())
 }
 
-// sortQueues returns a sorted shallow copy of the queues for this parent queue.
+// getSortedQueues returns a sorted shallow copy of the queues for this parent queue.
 // Only queues with a pending resource request are considered. The queues are sorted using the
 // sorting type for the parent queue.
 // Lock free call all locks are taken when needed in called functions
-func (sq *Queue) sortQueues() []*Queue {
+func (sq *Queue) getSortedQueues() []*Queue {
 	if sq.IsLeafQueue() {
 		return nil
 	}
+
+	sq.RLock()
```

Comment:
Extract this to a separate internal method, that `defer` right before the return doesn't make any sense.
Like:

```
func (sq *Queue) getCachedSortedQueues() []*Queue {
  sq.RLock()
  defer sq.RUnlock()
  
  return sq.sortedQueues
}
```